src/structs/GraphVertex.jl

```diff
@@ -0,0 +1,101 @@
+# each of these draw_* functions return functionsn with specified change keywords like radius, border_color etc.
+# expose as many of these props as supported by Luxor drawing
+node2 = @Graph(ga, 50:100, GraphVertex(2, draw; animate_on=:scale, property_style_map=Dict(:weight=>:radius)))
+render(video; pathname="graph_node.gif")
+```
+"""
+struct GraphVertex
+    node::Int
+    animate_on::Symbol
+    property_style_map::Dict{Any,Symbol}
+    opts::Dict{Symbol, Any}
+end
+
+"""
+    GraphVertex(node::Int, draw::Function; <keyword arguments>)
+    GraphVertex(graph::AbstractObject, node::Int, draw::Function; <keyword arguments>)
+    GraphVertex(graph::AbstractObject, node::Int, draw::Vector{Function}; <keyword arguments>)
+
+Create a graph node, specifying a drawing function or a property style map or both.
+
+# Arguments
+- `graph::AbstractObject`: The graph created using [`GraphAnimation`](@ref) to which this node should be added to.
+- `node::Int`: A unique id representing the node being added to the graph.
+    - Currently, only numeric node ids are supported.
+- `draw::Function`: The drawing function used to draw the node.
+    - Implementing the drawing function in a special way to expose the drawing parameters helps in better animation.
+
+# Keywords
+- `animate_on::Symbol`: Control the animation effect on the nodes using pre-defined drawing parameters. Default is `:opacity`.
+    - For nodes, it can be 
+        - `:opacity`
+        - `:scale`
+    - For known graph types, it can additionally be 
+        - `:fill_color`
+        - `:border_color`
+        - `:radius`
+- `property_style_map::Dict{Any,Symbol}`: A mapping to of how node attributes map to node drawing styles.
+"""
+GraphVertex(node::Int, draw::Union{Vector{Function}, Function}; kwargs...) =
+    GraphVertex(CURRENT_GRAPH[1], node, draw; kwargs...)
+
+GraphVertex(graph::AbstractObject, node::Int, draw::Vector{Function}; kwargs...) =
+    GraphVertex(graph, node, compile_draw_funcs(draw); kwargs...)
+
+function GraphVertex(
+    graph::AbstractObject,
+    node::Int,
+    draw::Function;
+    animate_on::Symbol = :opacity,
+    property_style_map::Dict{Any,Symbol} = Dict{Any,Symbol}(),
+)
+    g = graph.meta
+    if !(typeof(g) <: JGraph)
+        throw(ErrorException("Cannot define node since $(typeof(graph)) is not a `JGraph`"))
+    end
+    if g.mode == :static
+        if get_prop(g.adjacency_list, node) !== nothing
+            @warn "Node $(node) is already created on canvas. Recreating it will leave orphan node objects in the animation. To undo, call `rem_node!`"
+        end
+        if g.layout != :none
+            draw_fn = (args...; position=O, kwargs...) -> begin Luxor.translate(position); draw(args...; position=position, kwargs...) end
+        else
+            draw_fn = draw
+        end
+        add_vertex!(g.adjacency_list.graph)
+        set_prop!(g.adjacency_list, nv(g.adjacency_list), length(g.ordering)+1)
```

Comment:
Yes, to store more than one property for nodes and edges there is already the `MetaGraph` package. In many graph algorithms, only a single weight is associated with the nodes and edges hence this would be a useful structure to have.